Ensure libseccomp is installed before starting containerd on CentOS 8

OwenTuz opened the pull request:
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
Starting containerd (and therefore docker, which requires containerd) currently fails on Centos 8 due to a missing dependency. This PR copies what we already do for CRI-O to ensure libseccomp is installed.

To test, create `vagrant/config.rb` containing the line `$os = "centos8"` and provision, as described in issue #6920 

**Which issue(s) this PR fixes**:
Fixes #6920

**Special notes for your reviewer**:
Please flag if there's any way I can introduce better tests for this.

I had hoped to make sure it was covered, but it seems molecule only tests against ubuntu (reasonably enough, since it's meant to be fast), and I was unclear on whether CI currently tests against Centos 8.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

OwenTuz commented:
Re-pushed to fix invalid commit message

floryut commented:
@OwenTuz now that #6910 is merged, could you please rebase master ? Thank you!

OwenTuz commented:
@floryut done! Thanks for the heads up

OwenTuz commented:
Build failure looks like we've hit a rate limit on DockerHub:
```
fatal: [instance-1 -> None]: FAILED! => {"attempts": 4, "changed": true, "cmd": ["/usr/bin/docker", "pull", "docker.io/library/nginx:1.19"], "delta": "0:00:01.132849", "end": "2020-11-23 09:58:51.061621", "msg": "non-zero return code", "rc": 1, "start": "2020-11-23 09:58:49.928772", "stderr": "Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit", "stderr_lines": ["Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit"], "stdout": "", "stdout_lines": []}
```
https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/-/jobs/866416207

karlism commented:
@OwenTuz, wouldn't it be better idea instead of `command` module to use `yum`, `dnf` or `package` module with `state: latest`?

OwenTuz commented:
@karlism that's totally reasonable, this is almoost entirely a copy/paste job on my part. I can update and push?

OwenTuz commented:
Simplified this quite a bit as suggested by @karlism. Some notes:
- I've removed the version check that was in the previous commands. I hope I'm not missing a good reason for introducing this but I think `latest` works better as suggested. I'm also not sad to see the big chain of regexes disappear.


- Other packages are installed from the array variable `{docker,containerd,cri-o}-packages`, and it may be more consistent to add libseccomp to these. I've avoided this for now as I don't think the extra logic required (to add one package, only when using CentOS 8) helps readability.
However I think it's a matter of code style, so I'll flag it up here and am happy to change if it's preferable.

OwenTuz commented:
Found out why the version check was present! cri-o needs a version of libseccomp > 2.3. Docker and containerd don't seem to care.

I've reverted this to use `state=latest` and added a `noqa` comment for the corresponding ansible-lint rule, as I think it's probably the cleanest option here. Just testing and I'll push if all goes well.

LuckySB commented:
/ok-to-test

LuckySB commented on roles/container-engine/containerd/tasks/main.yml:
```diff
@@ -129,4 +129,13 @@
+    name: libseccomp
+    state: latest
+  when:
+    - ansible_distribution == "CentOS"
```
but what about the RedHat or Fedora ?
do they need this package too?

OwenTuz replied:
This is also copied and pasted from the CRI-O code: but I can say I haven't had this problem when testing with Fedora. Some other issues blocked me from investigating RHEL8, but I'll see if I can get past those and let you know.

OwenTuz replied:
@LuckySB Now that I've managed to get a developer subscription and test: I can confirm that RHEL8 and Fedora do not seem to need this package explicitly installed.

Seems my RHEL issues had been encountered before and fixed but there was a typo, I'll raise another PR :)

champtar replied:
chrony depends on libseccomp, that is why, please make it for all RedHat family (EL7/EL8)

oomichi commented:
/lgtm

LuckySB commented:
/approve

champtar commented:
@LuckySB you completely ignored my review ...

OwenTuz commented:
Sorry for not responding earlier, I was planning on taking a look tomorrow
- just saw the emails that this has merged.

I'm happy to raise another PR to address these comments if necessary.

Re `latest` vs `present` - I used `latest` here because `present` installs
the wrong version for CRI-O. Perhaps there's a better way to work around
this?

I'm probably missing something obvious, sorry, but I'm not entirely clear
on the objection to using `latest` here?

I've no problem with installing libseccomp for all the Red Hat variants.

On Thu, 3 Dec 2020, 21:44 Etienne Champetier, <notifications@github.com>
wrote:

> @LuckySB <https://github.com/LuckySB> you completely ignored my review ...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes-sigs/kubespray/pull/6922#issuecomment-738337590>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABYSNPAZ2PNPNWF5JTRYUG3STABCXANCNFSM4TY4ZGEA>
> .
>

champtar commented:
@OwenTuz no problem, I just commented today.
Re running `cluster.yml` on a cluster 1 month after first deployment should not change anything IMO (don't think it's the case right now)
Now looking at the changelog: https://centos.pkgs.org/8/centos-baseos-x86_64/libseccomp-2.4.1-1.el8.x86_64.rpm.html
CentOS 8 has 2.4.1-1 since 2019-05-31, so we could just drop this part for CRIO and be fine

Docker and containerd don't need latest, so lets not put present there

floryut commented:
Yup that's an approval a bit quick... sorry about that, maybe raise another PR indeed to address the issues/comments raise by @champtar

OwenTuz commented:
Sorry, things got busy elsewhere: just a note to say I've not forgotten abut this and will raise that PR in the next few days.
